include/xtensor/xsort.hpp

```diff
@@ -144,6 +144,163 @@ namespace xt
         return sort(de, de.dimension() - 1);
     }
 
+    namespace detail {
+        template <class VT, class T>
+        struct rebind_value_type
+        {
+            using type = xarray<VT, xt::layout_type::dynamic>;
```

Comment:
I think it might be better to use `XTENSOR_DEFAULT_LAYOUT` (which is row_major if not defined otherwise)?